packages/reporters/hmr-server/src/HMRReporter.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import {Reporter} from '@parcel/plugin';
+import HMRServer from './HMRServer';
+
+let hmrServer: HMRServer | null = null;
+export default new Reporter({
+  async report(event, options) {
+    if (options.hot) {
+      if (!hmrServer) {
+        hmrServer = new HMRServer();
+
+        if (options.hot === true) {
+          options.hot = {
```

Comment:
this will rewrite the same `ParcelOptions` object that's passed around the internals of Parcel and out to other plugins. Maybe just create a local and modify that?

I wonder if we be handing a deep copy of parcel options to plugins 🤔

Reply:
To be safe we probably would want to clone config before passing.

Reply:
One issue with this though, HMRServer should be able to pass a port to HMRRuntime in some way, not sure how to pass that.

Reply:
I was thinking about letting core normalise some options like serve and hot, but this would be adding extra complexity to parcel core that can probably be worked around. Also this wouldn't event resolve the port issue as that one's currently just wathever port is available.